android/app/src/main/java/com/zulipmobile/notifications/NotifyReact.kt

```diff
@@ -68,13 +68,15 @@ private fun emitOrLaunch(application: ReactApplication, eventName: String, data:
     val lifecycleState = reactContext.lifecycleState!!
     Log.d(TAG, "emitOrLaunch: lifecycle state is $lifecycleState")
     when (lifecycleState) {
-        LifecycleState.BEFORE_CREATE,
+        LifecycleState.BEFORE_CREATE -> {
             // Main activity has not yet "resumed" for the first time,
             // or else has been "destroyed" and will never be resumed again.
-        LifecycleState.BEFORE_RESUME
+            launchMainActivity(application as Context)
+        }
+        LifecycleState.BEFORE_RESUME -> {
             // Main activity has "resumed" (gone to foreground) at least once,
             // but is out of foreground now, but might come back.
-        -> {
+            emit(reactContext, eventName, data)
             launchMainActivity(application as Context)
         }
```

Comment:
> especially since (if I'm reading this correctly) the value of `NotificationsModule.initialNotification` isn't used in the branches where an event is emitted, so there's little reason to set it unconditionally.

Hmm, a good point! I agree that the `initialNotification` value shouldn't be having an effect in the cases where we're using `emit`.

I have an idea how to structure this better; will take a look.

Reply:
Updated! Please take a look.

Reply:
I like the final result much more, although some of the intermediate commits now meander down a misleading detour. (You may want to consider emending them, though I won't press for it.)

Reply:
Yeah... the branch probably could be shorter if revised start-to-finish with the new endpoint in mind. I considered that when revising yesterday, but decided the work required for that would mean taking our polishing of commits a step or two too far. :-)

I think it is at least the case that in this version each commit takes the code *closer* to the endpoint of the branch, which helps.